protected/humhub/modules/admin/controllers/PendingRegistrationsController.php

```diff
@@ -56,103 +40,73 @@ public function getAccessRules()
         ];
     }
 
-    public function actionIndex($export = false, $format = null)
+    public function actionIndex()
     {
         $searchModel = new PendingRegistrationSearch();
         $dataProvider = $searchModel->search(Yii::$app->request->queryParams);
 
-        if($export) {
-            return $this->createCVS($dataProvider, $searchModel, $format);
-        }
-
-
         return $this->render('index', [
             'dataProvider' => $dataProvider,
             'searchModel' => $searchModel,
-            'types' => $this->getTypeMapping()
+            'types' => [
+                null => null,
+                PendingRegistrationSearch::SOURCE_INVITE => Yii::t('AdminModule.base', 'Invite'),
+                PendingRegistrationSearch::SOURCE_SELF => Yii::t('AdminModule.base', 'Sign up'),
+            ]
         ]);
     }
 
-    public function getTypeMapping()
+    /**
+     * Export user list as csv or xlsx
+     * @param string $format supported format by phpspreadsheet
+     * @return \yii\web\Response
+     * @throws \PhpOffice\PhpSpreadsheet\Exception
+     * @throws \PhpOffice\PhpSpreadsheet\Writer\Exception
+     * @throws \yii\base\InvalidConfigException
+     */
+    public function actionExport($format)
     {
-        return [
-            PendingRegistrationSearch::SOURCE_INVITE => Yii::t('AdminModule.base', 'Invite'),
-            PendingRegistrationSearch::SOURCE_SELF => Yii::t('AdminModule.base', 'Sign up'),
-        ];
+        $searchModel = new PendingRegistrationSearch();
+        $dataProvider = $searchModel->search(Yii::$app->request->queryParams);
+
+        $exporter = new SpreadsheetExport([
+            'dataProvider' => $dataProvider,
+            'columns' => $this->collectExportColumns(),
+            'resultConfig' => [
+                'fileBaseName' => 'humhub_user',
+                'writerType' => $format,
+            ],
+        ]);
+
+        return $exporter->export()->send();
     }
 
-    public function createCVS($dataProvider, ActiveRecord $model, $format = null)
```

Comment:
BC break

Reply:
Thank you! We should add a note to changelog that createCVS is removed.